### PR TITLE
use filepath walk for flatfs query

### PR DESCRIPTION
**Pull request by whyrusleeping**
running a query on a large ipfs datastore takes forever (and uses a lot of ram), this should help, some.

### CODE REVIEW

File: flatfs/flatfs.go [hunk 1]
```diff
@@ -314,52 +315,33 @@ func (fs *Datastore) Query(q query.Query) (query.Results, error) {
...
-	for _, fi := range prefixes {
-		var err error
-		res, err = fs.enumerateKeys(fi, res)
-		if err != nil {
-			return nil, err
-		}
-	}
-	return query.ResultsWithEntries(q, res), nil
-}
+			if !info.Mode().IsRegular() || info.Name()[0] == '.' {
```
Comment (jbenet) [inline flatfs/flatfs.go:323]: `.` but no `..`?

**Comment by jbenet** [comment]
this LGTM

File: flatfs/flatfs.go (hunk 1 above)
Comment (whyrusleeping) [inline flatfs/flatfs.go:323]: I think the intent of this check was to ignore the random dotfiles that os'es (OSX) will randomly decide to put in folders. Other than that, i'm not certain.

Comment (jbenet) [inline flatfs/flatfs.go:323]: Oh name[0], we're good.

**Pull request merged**